src/org/wavescale/sourcesync/ui/ConnectionConfigPanel.java

```diff
@@ -107,7 +107,11 @@ public void setConnectionType(String connectionType) {
      * @return an int value representing the target port.
      */
     public int getPort() {
-        return Integer.valueOf(tfPort.getText());
+        try {
+            return Integer.valueOf(tfPort.getText());
+        } catch (NumberFormatException e) {
+            return 22;
```

Comment:
First of all I should say thank you :) I was not careful enough with this one. I do have a little comment though. If you have time you can fix it now. If you don't I will accept this pull and I will fix it later. 

So I was thinking that maybe we should we return  22 only when the connection type is sftp or scp, otherwise 21 is the default (for ftp and ftps). As a TODO - I think  a better approach for this would be to validate sensitive fields (like host, port, username value, root value) to make sure they don't have improper values like empty ones, etc..
If we find such things, the save should not happen and an error should be displayed (either through an error dialog, or - and I think this is much more used nowadays - through a footer label in  the "sourcesync connection configuraion" window)

Of course, this approach will take a little more effort than just returning a default value and silently hide the fact that user did not fill all the necessary values.